lib/ProductOpener/Tags.pm

```diff
@@ -3539,6 +3541,39 @@ sub add_users_translations_to_taxonomy($) {
 
 }
 
+#Return the last occurence of the property if several are defined in ingredients.txt
+sub find_property_in_ingredient_tree($$$) {
+
+	my $tagtype = shift;
+	my $canon_tagid = shift;
+	my $property = shift;
+
+	while ($canon_tagid) 
+	{
+		if (exists $properties{$tagtype}{$canon_tagid})
+		{
+			foreach(keys $properties{$tagtype}{$canon_tagid})
+			{
+				if ($_ eq $property) 
+				{	
+					return "$properties{$tagtype}{$canon_tagid}{$_}\n";
+				}
+			}
+		}
+		if ($direct_parents{$tagtype}{$canon_tagid})
+		{
+			foreach(keys $direct_parents{$tagtype}{$canon_tagid}){$canon_tagid = $_;}
+		}
+		else
+		{
+			return 0;
```

Comment:
Maybe return undef instead of 0, there might be properties which value is 0.

Reply:
ok